coreclient/src/conversations/persistence.rs

```diff
@@ -138,8 +138,14 @@ impl Conversation {
     ) -> Result<(), rusqlite::Error> {
         for (conversation_id, timestamp) in mark_as_read_data.into_iter() {
             transaction.execute(
-                "UPDATE conversations SET last_read = ? WHERE conversation_id = ?",
-                params![timestamp, conversation_id],
+                "UPDATE conversations 
+                 SET last_read = CASE 
+                                    WHEN last_read IS NULL OR last_read < ? 
+                                    THEN ? 
+                                    ELSE last_read 
+                                 END 
```

Comment:
Having this check here doesn't hurt, but shouldn't that check instead be at the level of the caller, i.e. in `mark_as_read_debounced` before a timestamp in the HashMap of `conversation_timestamps` is changed?